File: fetch-chaos-selfhosted-bounty.py
#!/usr/bin/env python3

"""
Fetch self-hosted Chaos programs that pay bounties
Fetches from Chaos index and filters for non-known-platform bounty programs
"""

import requests
import json


def fetch_chaos_selfhosted_bounty():
    """Fetch self-hosted Chaos programs paying bounties"""
    
    headers = {"User-Agent": "Mozilla/5.0"}
    
    known_platforms = {
        "hackerone",
        "bugcrowd",
        "intigriti",
        "yeswehack",
        "hackenproof",
        "issuehunt",
        "bugbountydirectory"
    }
    
    try:
        print("[*] Fetching Chaos index from projectdiscovery.io...")
        res = requests.get("https://chaos-data.projectdiscovery.io/index.json", headers=headers)
        res.raise_for_status()
        
        data = res.json()
        print(f"[*] Retrieved {len(data)} total programs from Chaos")
        
        self_hosted = []
        for p in data:
            platform = p.get("platform", "").lower().strip()
            bounty = p.get("bounty", False)
            
            # Filter: not in known platforms AND has bounty
            if platform not in known_platforms and bounty is True:
                self_hosted.append({
                    "name": p.get("name"),
                    "program_url": p.get("program_url"),
                    "platform": platform,
                    "bounty": bounty,
                    "subdomains_count": p.get("count"),
                    "download_url": p.get("URL"),
                })
        
        # Confirm filter
        self_hosted = [p for p in self_hosted if p["bounty"] is True]
        
        # Save to file
        with open("chaos_selfhosted.json", "w") as f:
            json.dump(self_hosted, f, indent=2)
        
        print(f"[✓] Total bounty=True only: {len(self_hosted)}")
        print("\n[*] Sample self-hosted bounty programs:")
        for p in self_hosted[:10]:
            print(f"  {p['name'][:30]:30} | {p['program_url']}")
        
    except requests.exceptions.RequestException as e:
        print(f"[✗] Network error: {e}")
    except json.JSONDecodeError as e:
        print(f"[✗] JSON parse error: {e}")
    except Exception as e:
        print(f"[✗] Error: {e}")


if __name__ == "__main__":
    fetch_chaos_selfhosted_bounty()

File: fetch-chaos-selfhosted-bounty.js
#!/usr/bin/env node

/**
 * Fetch self-hosted Chaos programs that pay bounties
 * Fetches from Chaos index and filters for non-known-platform bounty programs
 */

const https = require('https');
const fs = require('fs');

const KNOWN_PLATFORMS = new Set([
    'hackerone',
    'bugcrowd',
    'intigriti',
    'yeswehack',
    'hackenproof',
    'issuehunt',
    'bugbountydirectory'
]);

function fetchFromApi(url) {
    return new Promise((resolve, reject) => {
        const options = {
            headers: {
                'User-Agent': 'Mozilla/5.0'
            }
        };

        https
            .get(url, options, (res) => {
                let data = '';

                res.on('data', (chunk) => {
                    data += chunk;
                });

                res.on('end', () => {
                    try {
                        resolve(JSON.parse(data));
                    } catch (err) {
                        reject(new Error(`JSON parse error: ${err.message}`));
                    }
                });
            })
            .on('error', (err) => {
                reject(err);
            });
    });
}

async function fetchChaosSelfhostedBounty() {
    try {
        console.log('[*] Fetching Chaos index from projectdiscovery.io...');

        const data = await fetchFromApi('https://chaos-data.projectdiscovery.io/index.json');
        console.log(`[*] Retrieved ${data.length} total programs from Chaos`);

        const selfHosted = [];

        for (const p of data) {
            const platform = (p.platform || '').toLowerCase().trim();
            const bounty = p.bounty === true;

            // Filter: not in known platforms AND has bounty
            if (!KNOWN_PLATFORMS.has(platform) && bounty) {
                selfHosted.push({
                    name: p.name,
                    program_url: p.program_url,
                    platform: platform,
                    bounty: bounty,
                    subdomains_count: p.count,
                    download_url: p.URL
                });
            }
        }

        // Confirm filter
        const filtered = selfHosted.filter(p => p.bounty === true);

        // Save to file
        fs.writeFileSync('chaos_selfhosted.json', JSON.stringify(filtered, null, 2));

        console.log(`[✓] Total bounty=True only: ${filtered.length}`);
        console.log('\n[*] Sample self-hosted bounty programs:');
        filtered.slice(0, 10).forEach((p) => {
            console.log(`  ${p.name.padEnd(30)} | ${p.program_url}`);
        });

    } catch (error) {
        console.error(`[✗] Error: ${error.message}`);
        process.exit(1);
    }
}

fetchChaosSelfhostedBounty();